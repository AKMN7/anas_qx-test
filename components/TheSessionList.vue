<template>
  <div v-if="sessions.length" class="w-full space-y-3">
    <div
      v-for="session in sessions"
      :key="session.company_name"
      class="keep-ltr cursor-pointer rounded-xl border border-lightPurpleGrey p-3 text-base-color"
      :class="{
        selectedSession: session.company_name === selectedSession,
      }"
      @click="showBtns(session.company_name)"
    >
      <div class="space-2 flex items-center justify-start">
        <img
          :src="session.company_logo"
          alt="Logo"
          class="h-10 w-10 rounded-xl border border-borderqx"
        />
        <div class="flex w-fit flex-col py-1 px-2">
          <p class="text-sm font-bold">{{ session.company_name }}</p>
          <p>{{ session.company_description }}</p>
        </div>
      </div>

      <div class="mt-1 flex items-center justify-start">
        <p class="text-lightPurple">
          â†’ {{ session.start_time }} to
          {{ session.end_time }}
        </p>
      </div>

      <div class="cta-btns mt-3 flex items-center justify-start">
        <button
          class="rounded-xl bg-purple py-2 px-4 text-white"
          @click="goToLink(session.zoom_link)"
        >
          Join
        </button>
        <button class="border- ml-3 rounded-xl py-2 px-4 hover:font-bold">
          Dismiss
        </button>
      </div>
    </div>
  </div>

  <!-- Display No Sessions Illus if no sessions in the selected day -->
  <div
    v-else
    class="flex w-full flex-col items-center justify-center space-y-3 rounded-xl border border-lightPurpleGrey p-5 text-center"
  >
    <img
      src="~/assets/no-sessions.svg"
      alt="No_Data"
      class="h-45 w-44 lg:h-80 lg:w-80"
    />
    <p class="w-[200px] text-sm lg:w-[40%]">
      {{ $t('No-Sessions-0') }}
      <span class="font-bold">{{ $t('No-Sessions-1') }}</span>
      {{ $t('No-Sessions-2') }}.
    </p>
  </div>
</template>

<script>
export default {
  props: {
    sessions: {
      type: Array,
      required: true,
      default: () => [],
    },
  },
  data() {
    return {
      selectedSession: null,
    }
  },
  methods: {
    showBtns(name) {
      this.selectedSession = name
    },
    goToLink(link) {
      window.open(link)
    },
  },
}
</script>

<style scoped>
.selectedSession {
  background-color: rgba(148, 132, 247, 0.08);
}

.selectedSession .cta-btns {
  display: block !important;
}

.cta-btns {
  display: none;
}
</style>
