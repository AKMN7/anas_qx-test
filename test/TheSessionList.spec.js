import { mount } from '@vue/test-utils'
import TheSessionList from '../components/TheSessionList.vue'

describe('Index', () => {
  let cmp

  beforeEach(() => {
    cmp = mount(TheSessionList, {
      // props is overridden using `propsData`
      propsData: {
        sessions: ['Test Sessions'],
      },
    })
  })

  it('has received ["Test Sessions"] as the sessions property', () => {
    expect(cmp.vm.sessions).toEqual(['Test Sessions'])
  })

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  })
})
