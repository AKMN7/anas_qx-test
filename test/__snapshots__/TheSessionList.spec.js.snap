// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index has the expected html structure 1`] = `
<div
  class="keep-ltr grid w-full grid-cols-1 gap-3 md:grid-cols-2 lg:grid-cols-3"
>
  <div
    class="keep-ltr session h-fit w-full cursor-pointer rounded-xl border border-lightPurpleGrey p-3 text-base-color"
  >
    <div
      class="space-2 flex items-center justify-start"
    >
      <img
        alt="Logo"
        class="h-10 w-10 rounded-xl border border-borderqx"
      />
       
      <div
        class="flex w-fit flex-col py-1 px-2"
      >
        <p
          class="text-sm font-bold"
        >
          
        </p>
         
        <p>
          
        </p>
      </div>
    </div>
     
    <div
      class="mt-1 flex items-center justify-start"
    >
      <p
        class="text-unselected"
      >
        
        â†’  to
        
      
      </p>
    </div>
     
    <div
      class="cta-btns mt-3 flex items-center justify-start"
    >
      <button
        class="rounded-xl bg-purple py-2 px-4 text-white"
      >
        
        Join
      
      </button>
       
      <button
        class="border- ml-3 rounded-xl py-2 px-4 hover:font-bold"
      >
        
        Dismiss
      
      </button>
    </div>
  </div>
</div>
`;
